packages/back-end/src/jobs/updateExperimentResults.ts

```diff
@@ -237,8 +239,8 @@ async function updateSingleExperiment(job: UpdateSingleExpJob) {
     // If we failed to update the experiment, turn off auto-updating for the future
     try {
       experiment.autoSnapshots = false;
-      experiment.markModified("autoSnapshots");
-      await experiment.save();
+      changes.autoSnapshots = false;
+      await updateExperimentById(experiment, changes);
```

Comment:
In this case, you can remove the `changes` variable entirely and just make this call inline:

```ts
await updateExperimentById(experiment, { autoSnapshots: false })
```

The way the code is currently, it implies that somewhere else in the ~100 lines above is also adding changes to the experiment, but that's not the case here.